src/effects/VST/VSTEffect.cpp

```diff
@@ -1308,6 +1308,10 @@ std::unique_ptr<EffectUIValidator> VSTEffect::PopulateUI(ShuttleGui &S,
    }
 
    auto pParent = S.GetParent();
+
+   // Looks like pushing the event handler is still needed, otherwise
+   // an assert will pop up when closing the gui:
+   // "xwWindowsBase::PopEventHandler(): cannot pop the wxWindows itself"
```

Comment:
The solution for this:  stop inheriting `DefaultEffectUIValidator`, just inherit `EffectUIValidator`

Which means you no longer have the overrides of `ValidateUI`, `IsGraphicalUI`, `Disconnect` working for you.

But the first is already overridden in your class by the time we reach this commit, `IsGraphicalUI` is trivial to do, and `Disconnect` is unimportant (you could override it if you like, to zero out the parent and dialog pointers)

Reply:
thanks for the tip, done